bl/board/inc/board.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <type_traits>
+
+#include "block.hpp"
+#include "drawable.hpp"
+
+class Board {
+   private:
+    template <typename>
+    struct is_tuple : std::false_type {};
+    template <typename pos, typename ptr>
+    struct is_tuple<std::tuple<pos, pos, ptr>> : std::true_type {};
+
+   public:
+    template <
+        class... Args, std::enable_if_t<(is_tuple<Args>::value && ...), bool> = true,
```

Comment:
Very nice template magic, but its is completely unnecessary over complicated. You could just do
addBlocks(Block* ... block)

Reply:
if i do addBlocks(Block*  block...) I cannot use fold expression and have to use va_start, va_end and va_arg and do it C-style way. Is it better?

Reply:
ok, good poitnt, so lets do it like that 

```cpp
template<typename ... Args>
void add_block<tuple<Block*,int,int> block, Args... blocks>
{
blocks.push_back( {Drawable::Position{std::get<0>(blck), std::get<1>(blck)}, std::get<2>(blck)}),
 add_block(block...)
}
```

Reply:
Oh, you can also just do 
```cpp
void add_block(std::initializer_list<tuple<Block*,int,int>> blocks) 
{

}
```
And you don't have to worry about passed types :)

Reply:
Automatic template parameter deduction does not work, I would need to explicitly say what parameters are given (i.e std::tuple<int, int, std::shared_ptr<Block>>{1, 1, p} instead of {1, 1, p}) so I changed this function to add just 1 block :(

Reply:
ok, if you say so :) lets leave it as is

Reply:
didnt expect you checking it at this hour so i wrote comments before pushing changes :P my bad